### ISSUE TITLE
[ArchLinux] error: could not determine root mount point /

### ISSUE THREAD

**Issue by z3ntu** [openSUSE/obs-build#330]
```
luca@g550jk /m/h/h/razer-drivers> osc build Arch_Extra x86_64 PKGBUILD 
Run source service: /usr/lib/obs/service/download_url --protocol https --host github.com --path /terrycain/razer-drivers/archive/v1.1.8.tar.gz --outdir /mnt/hdd/home:z3ntu/razer-drivers/tmpGiM8Ap.download_url.service
Run source service: /usr/lib/obs/service/verify_file --file _service:download_url:v1.1.8.tar.gz --verifier sha256 --checksum b79df8ae7c1060c4656d1a70ddb3b2178ed9fd0708dfbc9308b539382891bc6e --outdir /mnt/hdd/home:z3ntu/razer-drivers/tmpYaKmzz.verify_file.service
Building PKGBUILD for Arch_Extra/x86_64
Getting buildinfo from server and store to /mnt/hdd/home:z3ntu/razer-drivers/.osc/_buildinfo-Arch_Extra-x86_64.xml
Getting buildconfig from server and store to /mnt/hdd/home:z3ntu/razer-drivers/.osc/_buildconfig-Arch_Extra-x86_64
Updating cache of required packages
0.0% cache miss. 100/100 dependencies cached.

WARNING: unknown packages get not verified, they can compromise your system !
Writing build configuration
Running build
[sudo] password for luca: 
VM_IMAGE: , VM_SWAP: 
logging output to /var/tmp/build-root/Arch_Extra-x86_64/.build.log...
[    0s] Memory limit set to 32941444KB
[    0s] Using BUILD_ROOT=/var/tmp/build-root/Arch_Extra-x86_64
[    0s] Using BUILD_ARCH=x86_64:i686:i586:i486:i386
[    0s] 
[    0s] 
[    0s] g550jk started "build PKGBUILD" at Tue Jan 31 06:29:32 UTC 2017.
[    0s] 
[    0s] 
[    0s] processing specfile /mnt/hdd/home:z3ntu/razer-drivers/PKGBUILD ...
[    0s] running changelog2spec --target rpm --file /mnt/hdd/home:z3ntu/razer-drivers/PKGBUILD
[    0s] init_buildsystem --configdir /usr/lib/build/configs --cachedir /var/cache/build --rpmlist /tmp/rpmlist.h1wqIs /mnt/hdd/home:z3ntu/razer-drivers/PKGBUILD ...
[    0s] It seems that there was an incomplete setup of /var/tmp/build-root/Arch_Extra-x86_64.
[    0s] To be sure, we will build it again completely...
[    0s] Your build system is broken!! Shall I execute
[    0s] 
[    0s]     rm -rf -- /var/tmp/build-root/Arch_Extra-x86_64/*
[    0s] 
[    0s] y - yes, cleanup the build root
[    0s] N - No, abort build (default on enter)
[    0s] c - Continue anyway with this build root
y
[    2s] [y/N/c] [1/35] preinstalling filesystem...
[    2s] [2/35] preinstalling pacman-mirrorlist...
[    2s] [3/35] preinstalling glibc...
[    2s] [4/35] preinstalling attr...
[    2s] [5/35] preinstalling bzip2...
[    2s] [6/35] preinstalling expat...
[    2s] [7/35] preinstalling gcc-libs...
[    4s] [8/35] preinstalling grep...
[    4s] [9/35] preinstalling libidn...
[    4s] [10/35] preinstalling lz4...
[    4s] [11/35] preinstalling lzo...
[    4s] [12/35] preinstalling perl...
[    5s] [13/35] preinstalling zlib...
[    5s] [14/35] preinstalling acl...
[    5s] [15/35] preinstalling openssl...
[    5s] [16/35] preinstalling ncurses...
[    5s] [17/35] preinstalling libssh2...
[    5s] [18/35] preinstalling readline...
[    5s] [19/35] preinstalling sed...
[    5s] [20/35] preinstalling coreutils...
[    5s] [21/35] preinstalling bash...
[    5s] [22/35] preinstalling e2fsprogs...
[    5s] [23/35] preinstalling xz...
[    5s] [24/35] preinstalling gawk...
[    6s] [25/35] preinstalling gzip...
[    6s] [26/35] preinstalling icu...
[    6s] [27/35] preinstalling keyutils...
[    6s] [28/35] preinstalling libgpg-error...
[    6s] [29/35] preinstalling gpgme...
[    6s] [30/35] preinstalling libassuan...
[    6s] [31/35] preinstalling libpsl...
[    6s] [32/35] preinstalling krb5...
[    6s] [33/35] preinstalling libarchive...
[    6s] [34/35] preinstalling curl...
[    6s] [35/35] preinstalling pacman...
[    7s] 
[    7s] reordering...done
[    7s] [1/100] installing archlinux-keyring-20170104-1-any
[    7s] looking for conflicting packages...
[    7s] Packages (1) archlinux-keyring-20170104-1
[    7s] :: Proceed with installation? [Y/n] 
[    7s] checking keyring...
[    7s] checking for file conflicts...
[    7s] error: could not determine root mount point /
[    7s] error: not enough free disk space
[    7s] error: failed to commit transaction (not enough free disk space)
[    7s] Errors occurred, no packages were upgraded.
[    7s] exit ...

The buildroot was: /var/tmp/build-root/Arch_Extra-x86_64
```
```
$ df -h
Filesystem              Size  Used Avail Use% Mounted on
dev                     7,8G     0  7,8G   0% /dev
run                     7,8G  1,4M  7,8G   1% /run
/dev/mapper/MyVol-root  123G   41G   77G  35% /
tmpfs                   7,8G  229M  7,6G   3% /dev/shm
tmpfs                   7,8G     0  7,8G   0% /sys/fs/cgroup
tmpfs                   7,8G  2,1M  7,8G   1% /tmp
/dev/sda1               511M   79M  433M  16% /boot
/dev/mapper/MyVol-home   98G   35G   58G  38% /home
tmpfs                   2,0G  228M  1,8G  12% /home/luca/.cache
/dev/sdb1               902G  612G  244G  72% /mnt/hdd
tmpfs                   1,6G   92K  1,6G   1% /run/user/1000
```
Maybe related to #222

**Comment by illuusio** [comment]
Can you check if that CheckSpace is changed to another name and it's false or it there that config parameter anymore because Arch is fast moving target.

**Comment by z3ntu** [comment]
According to https://www.archlinux.org/pacman/pacman.conf.5.html there is still a CheckSpace option in the pacman.conf.

**Comment by z3ntu** [comment]
The `pacman` command with debug output enabled (in `/usr/lib/build/init_buildsystem`):
```
[    8s] debug: pacman v5.0.1 - libalpm v10.0.1
[    8s] debug: config: attempting to read file /etc/pacman.conf
[    8s] debug: config: new section 'options'
[    8s] debug: config: HoldPkg: pacman
[    8s] debug: config: HoldPkg: glibc
[    8s] debug: config: arch: x86_64
[    8s] debug: config: SigLevel: Required
[    8s] debug: config: SigLevel: DatabaseOptional
[    8s] debug: config: LocalFileSigLevel: Optional
[    8s] debug: config: new section 'core'
[    8s] debug: config file /etc/pacman.conf, line 76: including /etc/pacman.d/mirrorlist
[    8s] debug: config: new section 'extra'
[    8s] debug: config file /etc/pacman.conf, line 79: including /etc/pacman.d/mirrorlist
[    8s] debug: config: new section 'community'
[    8s] debug: config file /etc/pacman.conf, line 85: including /etc/pacman.d/mirrorlist
[    8s] debug: config: finished parsing /etc/pacman.conf
[    8s] debug: setup_libalpm called
[    8s] debug: option 'logfile' = /var/log/pacman.log
[    8s] debug: option 'gpgdir' = /etc/pacman.d/gnupg/
[    8s] debug: option 'hookdir' = /etc/pacman.d/hooks/
[    8s] debug: option 'cachedir' = /var/cache/pacman/pkg/
[    8s] debug: registering sync database 'core'
[    8s] debug: database path for tree core set to /var/lib/pacman/sync/core.db
[    8s] debug: "/var/lib/pacman/sync/core.db.sig" is not readable: No such file or directory
[    8s] debug: sig path /var/lib/pacman/sync/core.db.sig could not be opened
[    8s] debug: missing optional signature
[    8s] debug: setting usage of 15 for core repository
[    8s] debug: registering sync database 'extra'
[    8s] debug: database path for tree extra set to /var/lib/pacman/sync/extra.db
[    8s] debug: "/var/lib/pacman/sync/extra.db.sig" is not readable: No such file or directory
[    8s] debug: sig path /var/lib/pacman/sync/extra.db.sig could not be opened
[    8s] debug: missing optional signature
[    8s] debug: setting usage of 15 for extra repository
[    8s] debug: registering sync database 'community'
[    8s] debug: database path for tree community set to /var/lib/pacman/sync/community.db
[    8s] debug: "/var/lib/pacman/sync/community.db.sig" is not readable: No such file or directory
[    8s] debug: sig path /var/lib/pacman/sync/community.db.sig could not be opened
[    8s] debug: missing optional signature
[    8s] debug: setting usage of 15 for community repository
[    8s] debug: ".init_b_cache/archlinux-keyring.arch.sig" is not readable: No such file or directory
[    8s] debug: ".init_b_cache/archlinux-keyring.arch.sig" is not readable: No such file or directory
[    8s] debug: sig data: <from .sig>
[    8s] debug: ".init_b_cache/archlinux-keyring.arch.sig" is not readable: No such file or directory
[    8s] debug: sig path .init_b_cache/archlinux-keyring.arch.sig could not be opened
[    8s] debug: missing optional signature
[    8s] debug: opening archive .init_b_cache/archlinux-keyring.arch
[    8s] debug: starting package load for .init_b_cache/archlinux-keyring.arch
[    8s] debug: found mtree for package .init_b_cache/archlinux-keyring.arch, getting file list
[    8s] debug: finished mtree reading for .init_b_cache/archlinux-keyring.arch
[    8s] debug: sorting package filelist for .init_b_cache/archlinux-keyring.arch
[    8s] debug: adding package 'archlinux-keyring'
[    8s] debug: loading package cache for repository 'local'
[    8s] debug: added 0 packages to package cache for db 'local'
[    8s] debug: adding package archlinux-keyring-20170104-1 to the transaction add list
[    8s] looking for conflicting packages...
[    8s] debug: looking for conflicts
[    8s] debug: check targets vs targets
[    8s] debug: check targets vs targets
[    8s] debug: check targets vs db and db vs targets
[    8s] debug: check targets vs db
[    8s] debug: check db vs targets
[    8s] debug: returning error 32 from alpm_db_get_pkg : could not find or read package
[    8s] debug: returning error 32 from alpm_db_get_pkg : could not find or read package
[    8s] Packages (1) archlinux-keyring-20170104-1
[    8s] :: Proceed with installation? [Y/n] debug: using cachedir: /var/cache/pacman/pkg/
[    8s] debug: using cachedir: /var/cache/pacman/pkg/
[    8s] checking keyring...
[    8s] checking for file conflicts...
[    8s] debug: looking for file conflicts
[    8s] debug: searching for file conflicts: archlinux-keyring
[    8s] debug: searching for filesystem conflicts: archlinux-keyring
[    8s] debug: checking possible conflict: /usr/
[    8s] debug: file is a directory, not a conflict
[    8s] debug: checking possible conflict: /usr/share/
[    8s] debug: file is a directory, not a conflict
[    8s] debug: checking possible conflict: /usr/share/pacman/
[    8s] debug: file is a directory, not a conflict
[    8s] debug: checking available disk space
[    8s] debug: discovered mountpoint: /proc
[    8s] debug: discovered mountpoint: /dev/pts
[    8s] debug: discovered mountpoint: /dev/pts
[    8s] error: could not determine root mount point /
[    8s] debug: returning error 7 from _alpm_check_diskspace : not enough free disk space
[    8s] error: not enough free disk space
[    8s] error: failed to commit transaction (not enough free disk space)
[    8s] debug: unregistering database 'local'
[    8s] debug: freeing package cache for repository 'local'
[    8s] debug: unregistering database 'core'
[    8s] debug: unregistering database 'extra'
[    8s] debug: unregistering database 'community'
[    8s] Errors occurred, no packages were upgraded.
[    8s] exit ...
```

**Comment by z3ntu** [comment]
I think I have solved the mystery. The `CheckSpace` option is 

> * checkspace: Check disk space before installing.

which is exactly what we DON'T want here. So commenting the `CheckSpace` option in the `pacman.conf` solves the issue. I used `watch -n0.5 sudo sed -i 's/CheckSpace/#CheckSpace/' /var/tmp/build-root/Arch_Extra-x86_64/etc/pacman.conf` to comment the CheckSpace option.

**Comment by illuusio** [comment]
Nice catch! Did you do some PR or patch for this?

**Comment by z3ntu** [comment]
I believe, the `pacman.conf` is included in the OBS package for `pacman` (and I don't know where to patch this) ;)

**Comment by z3ntu** [comment]
So... I was using version 2.5.0 and only recently noticed that more stuff wasn't working properly anymore (eg debian building) and decided to use osc-build-git and noticed that it was fixed there (581a58984b5642e3d16f960ead9af85b6a1ba2f2). So it would be great if a new stable release could be made (as the last one was 2014) :)
